ash/src/extensions/ext/extended_dynamic_state.rs

```diff
@@ -0,0 +1,200 @@
+        &self,
+        command_buffer: vk::CommandBuffer,
+        cull_mode: vk::CullModeFlags,
+    ) {
+        self.extended_dynamic_state_fn
+            .cmd_set_cull_mode_ext(command_buffer, cull_mode);
+    }
+
+    #[doc = "<https://www.khronos.org/registry/vulkan/specs/1.2-extensions/man/html/vkCmdSetFrontFaceEXT.html>"]
+    pub unsafe fn cmd_set_front_face_ext(
+        &self,
+        command_buffer: vk::CommandBuffer,
+        front_face: vk::FrontFace,
+    ) {
+        self.extended_dynamic_state_fn
+            .cmd_set_front_face_ext(command_buffer, front_face)
+    }
+
+    #[doc = "<https://www.khronos.org/registry/vulkan/specs/1.2-extensions/man/html/vkCmdSetPrimitiveTopologyEXT.html>"]
+    pub unsafe fn cmd_set_primitive_topology_ext(
+        &self,
+        command_buffer: vk::CommandBuffer,
+        primitive_topology: vk::PrimitiveTopology,
+    ) {
+        self.extended_dynamic_state_fn
+            .cmd_set_primitive_topology_ext(command_buffer, primitive_topology)
+    }
+
+    #[doc = "<https://www.khronos.org/registry/vulkan/specs/1.2-extensions/man/html/vkCmdSetViewportWithCountEXT.html>"]
+    pub unsafe fn cmd_set_viewport_with_count_ext(
+        &self,
+        command_buffer: vk::CommandBuffer,
+        viewports: &[vk::Viewport],
+    ) {
+        self.extended_dynamic_state_fn
+            .cmd_set_viewport_with_count_ext(
+                command_buffer,
+                viewports.len() as u32,
+                viewports.as_ptr(),
+            )
+    }
+
+    #[doc = "<https://www.khronos.org/registry/vulkan/specs/1.2-extensions/man/html/vkCmdSetScissorWithCountEXT.html>"]
+    pub unsafe fn cmd_set_scissor_with_count_ext(
+        &self,
+        command_buffer: vk::CommandBuffer,
+        scissors: &[vk::Rect2D],
+    ) {
+        self.extended_dynamic_state_fn
+            .cmd_set_scissor_with_count_ext(
+                command_buffer,
+                scissors.len() as u32,
+                scissors.as_ptr(),
+            )
+    }
+
+    #[doc = "<https://www.khronos.org/registry/vulkan/specs/1.2-extensions/man/html/vkCmdBindVertexBuffers2EXT.html>"]
+    pub unsafe fn cmd_bind_vertex_buffers2_ext(
+        &self,
+        command_buffer: vk::CommandBuffer,
+        first_binding: u32,
+        binding_count: u32,
+        buffers: &[vk::Buffer],
+        offsets: &[vk::DeviceSize],
+        sizes: Option<&[vk::DeviceSize]>,
+        strides: Option<&[vk::DeviceSize]>,
+    ) {
+        assert!(buffers.len() >= binding_count as usize);
+        assert!(offsets.len() >= binding_count as usize);
+        let p_sizes = if let Some(sizes) = sizes {
+            assert!(sizes.len() >= binding_count as usize);
+            sizes.as_ptr()
+        } else {
+            ptr::null()
+        };
+        let p_strides = if let Some(strides) = strides {
+            assert!(strides.len() >= binding_count as usize);
+            strides.as_ptr()
+        } else {
+            ptr::null()
+        };
```

Comment:
I'd probably go for an exact match (`assert_eq!`), these are slices anyway that are very easy to construct and that way the user can't make any accidental mistakes. Then I'd just remove the `binding_count` argument altogether and derive that from the slices (that should all have equivalent length).

Reply:
Perhaps worth checking the other extensions, IIRC I've done it this way in the wrappers written by myself, not sure about the rest.

Reply:
Done.